File: client/src/app/estimate/models/customer.ts
import { _2d } from "./_2d";
import { _3d } from "./_3d";
import { item } from "./item";

export class Customer {
    public name = "";
    public phone = "";
    public _items: [item] = [new item()];

    public area = 0;
    public area_price = 0;
    public price = 0;
    public start_date !: string;
    public end_date !: string;
    public total_plywood_sheets!: number;
    public plywood_sheet_wastage!: number;
    constructor() { }
}

File: server/src/main/java/com/airtajal/model/customer.java
package com.airtajal.model;

import java.util.Set;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import java.io.Serializable;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class customer implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;

    String name, phone;
    Integer area, area_price, total_plywood_sheets, plywood_sheet_wastage;
    Long start_date, end_date;
    Set<item> items;
    Set<_3d> _3d_arr;

}
